src/studio-home/data/slice.js

```diff
@@ -29,15 +32,21 @@ const slice = createSlice({
       Object.assign(state.studioHomeData, payload);
     },
     fetchCourseDataSuccess: (state, { payload }) => {
-      const { courses, archivedCourses, inProcessCourseActions } = payload;
+      const { courses, archivedCourses = [], inProcessCourseActions } = payload.results;
+      const { numPages, count } = payload;
       state.studioHomeData.courses = courses;
       state.studioHomeData.archivedCourses = archivedCourses;
       state.studioHomeData.inProcessCourseActions = inProcessCourseActions;
+      state.studioHomeData.numPages = numPages;
+      state.studioHomeData.coursesCount = count;
     },
     fetchLibraryDataSuccess: (state, { payload }) => {
       const { libraries } = payload;
       state.studioHomeData.libraries = libraries;
     },
+    updateStudioHomeCoursesCustomParams: (state, { payload }) => {
+      state.studioHomeCoursesCustomParams = { ...state.studioHomeCoursesCustomParams, ...payload };
```

Comment:
Why do you make a copy for the previous state if you only send as a payload the currentPage?

Reply:
For now it's only currentPage but I will add more data here however I will delete it

Reply:
But in the case you add other element to the object you can directly modify it in the action after destructuring the object:
const {currentPage} = payload
state.studioHomeCoursesCustomParams.currentPage = currentPage
So it's not necessary make copies of the past state, that's the advantage of redux

Reply:
Let me check that shortly, thanks a lot